app/core/src/main/java/com/fsck/k9/helper/MessageHelper.kt

```diff
@@ -0,0 +1,129 @@
+package com.fsck.k9.helper
+
+import android.text.Spannable
+import android.text.SpannableString
+import android.text.SpannableStringBuilder
+import android.text.TextUtils
+import android.text.style.ForegroundColorSpan
+import com.fsck.k9.CoreResourceProvider
+import com.fsck.k9.DI
+import com.fsck.k9.K9.contactNameColor
+import com.fsck.k9.K9.isChangeContactNameColor
+import com.fsck.k9.K9.isShowContactName
+import com.fsck.k9.K9.isShowCorrespondentNames
+import com.fsck.k9.mail.Address
+import java.util.regex.Pattern
+
+class MessageHelper(
+    private val resourceProvider: CoreResourceProvider,
+) {
+
+    fun getSenderDisplayName(address: Address?): CharSequence {
+        if (address == null) {
+            return resourceProvider.contactUnknownSender()
+        }
+        val contactHelper = if (isShowContactName) DI.get(Contacts::class.java) else null
```

Comment:
Switching this to use constructor injection seems like a good thematic fit with removing `getInstance()`.